sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/HadoopFileDataObject.scala

```diff
@@ -81,16 +82,32 @@ private[smartdatalake] trait HadoopFileDataObject extends FileRefDataObject with
    * Check if the input files exist.
    * Note that hadoopDir can be a specific file or a directory.
    */
-  private[smartdatalake] def checkFilesExisting(implicit context: ActionPipelineContext): Boolean = {
+  def checkFilesExisting(implicit context: ActionPipelineContext): Boolean = {
     val status = try {
       filesystem.getFileStatus(hadoopPath)
     } catch {
       case _: FileNotFoundException => return false
     }
-    status.isFile || {
-      val globPath = if (partitions.nonEmpty) new Path(hadoopPath, PartitionValues(Map()).getPartitionString(partitionLayout().get)) else hadoopPath
-      status.isDirectory && filesystem.globStatus(new Path(globPath, fileName)).exists(_.isFile)
-    }
+    status.isFile || (status.isDirectory && listDataFiles().nonEmpty)
+  }
+
+  def listDataFiles(pv: PartitionValues = PartitionValues(Map()))(implicit context: ActionPipelineContext): Iterator[Path] = {
+    val pathPattern = if (partitions.nonEmpty) new GlobPattern(new Path(new Path(hadoopPath, pv.getPartitionString(partitionLayout().get)), fileName).toString)
+    else new GlobPattern(new Path(hadoopPath, fileName).toString)
+    RemoteIteratorWrapper(filesystem.listFiles(hadoopPath, partitions.nonEmpty /*recursive*/))
```

Comment:
I cannot see where the recursion is by looking at the code without IntelliJ... Will try in an IDE tomorrow

Reply:
This is not recursive, right? Can you remove the /*recursive*/ comment?